### ISSUE TITLE
Dis loss decays to zero while training oracle

### ISSUE THREAD

**Issue by aashiqmuhamed** [weilinie/RelGAN#18]
I was trying to get a better understanding of the behavior of g_loss and d_loss during adversarial training.

It appears that while using the oracle generated data, the gen_loss increases and dis_loss goes to zero. With the rsgan loss I thought both losses should converge to 0.7 when the net is trained. Since you haven't reported these losses in the paper, I'm curious what your thoughts are on balanced training.

**Comment by weilinie** [comment]
<img width="1420" alt="image" src="https://user-images.githubusercontent.com/25551679/84312763-7ef6b280-ab33-11ea-9e67-6a6c9a4b7bed.png">

Here are two loss plots for oracle data, where we may have different training behaviors depending on the (inverse) temperature. As you can see, the G training loss does not always go up while the D loss does not always decrease. Ideally, we expect GAN training could reach a theoretical “Nash Equilibrium” (as you mentioned 0.7 for both losses with the RSGAN loss), but in reality, it’s hard to say if the “Nash Equilibrium” exists or there is any guarantee that our current algorithms can reach this optimal point. In my opinion, we were instead seeking for a better suboptimal (or Pareto-like) solution.

**Comment by aashiqmuhamed** [comment]
Thanks, this was really helpful. I wonder if regularizing the discriminator could help achieve that equilibrium. Given that you've already achieved SOTA, perhaps this might help push it further. What are your thoughts?

**Comment by weilinie** [comment]
That’s right. I didn’t tune much regarding different regularization methods on the discriminator. I agree that in some cases where D quickly goes to zero, which means D is outracing G, properly regularizing D may achieve a better balance.
